unitTests/CommunicationPattern.C

```diff
@@ -24,75 +24,264 @@ protected:
+            send_counts[i] = num_elements*i;
+
+    std::vector<int> recv_counts(comm_size);
+
+    recv_counts[0] = num_elements*comm_rank;
+
+    std::vector<int> send_offsets(comm_size+1);
+    if (comm_rank == 0)
+        for (int i = 0; i < comm_size; i++)
+            send_offsets[i+1] = send_offsets[i] + num_elements*i;
+    
+    std::vector<int> recv_offsets(comm_size+1);
+    recv_offsets.back() = num_elements * comm_rank;
+
+    // Act
     auto comm_counts =
-        compute_gather_to_owner_counts(*exec.get(), 1, label(10));
-    // expected results
-    // if gathering to just one owner all 10 elements are send to it self
-    std::vector<int> send_counts(comm->size(), 0);
-    send_counts[comm->rank()] = 10;
-    std::vector<std::vector<int>> send_results(comm->size(), send_counts);
-    // all offsets should be zero
-    // last entry in offsets is total number of send elements
-    std::vector<int> send_offsets(comm->size() + 1, 0);
-    send_offsets.back() = 10;
-    std::vector<std::vector<int>> offset_results(comm->size(), send_offsets);
+        compute_scatter_from_owner_counts(*exec.get(), comm_size, label(num_elements * comm_rank)); 
+
+    // Assert
+    EXPECT_EQ(comm_counts.send_counts, send_counts);
+    EXPECT_EQ(comm_counts.recv_counts, recv_counts);
+    EXPECT_EQ(comm_counts.send_offsets, send_offsets);
+    EXPECT_EQ(comm_counts.recv_offsets, recv_offsets);
+}
+
+
+// TEST_F(CommunicationPatternFixture, compute_scatter_from_owner_counts_two_owners)
+// {
+//     // Arrange
+//     auto comm = exec->get_gko_mpi_host_comm();
+//     auto num_elements = 10;
+//     auto comm_size = comm->size();
+//     auto comm_rank = comm->rank();
+
+//     // Scatter different number of elements from owner to each non-owner process
+//     std::vector<int> send_counts(comm_size);
+    
+//     if (comm_rank == 0)
+//         for (int i = 1; i < comm_size/2; i++)
+//             send_counts[i] = num_elements*i;
+    
+//     if (comm_rank == comm_size/2)
+//         for (int i = comm_size/2 + 1; i < comm_size; i++)
+//             send_counts[i] = num_elements*i;
+      
+//     // std::vector<int> recv_counts(comm_size);
+
+//     // if (comm_rank != 0)
+//     //     recv_counts[0] = num_elements*comm_rank;
+
+//     // std::vector<int> send_offsets(comm_size);
+//     // if (comm_rank == 0)
+//     //     for (int i = 0; i < comm_size-1; i++)
+//     //         send_offsets[i+1] = send_offsets[i] + num_elements*i;
+    
+//     // std::vector<int> recv_offsets(comm_size);
+
+//     // Act
+//     auto comm_counts =
+//         compute_scatter_from_owner_counts(*exec.get(), comm_size/2, label(num_elements * comm_rank)); 
 
+//     // Assert
+//     // test send counts and revc counts
+//     EXPECT_EQ(comm_counts.send_counts, send_counts);
+//     // EXPECT_EQ(comm_counts.recv_counts, recv_counts);
+//     // EXPECT_EQ(comm_counts.send_offsets, send_offsets);
+//     // EXPECT_EQ(comm_counts.recv_offsets, recv_offsets);
+// }
```

Comment:
Please don't leave large chunks of uncommented code. Either implement or remove. (or leave a note why it is there)
```suggestion
```

Reply:
I should have mentioned that this pull request is not ready for the associated branch to be merged.  
I will add more unit tests